src/engine/GroupBy.cpp

```diff
@@ -1034,14 +1042,31 @@ concept SupportedAggregates =
     ad_utility::SameAsAnyTypeIn<A, GroupBy::AggregationDataVectors>;
 
 // _____________________________________________________________________________
-std::vector<size_t> GroupBy::HashMapAggregationData::getHashEntries(
-    std::span<const Id> ids) {
-  std::vector<size_t> hashEntries;
-  hashEntries.reserve(ids.size());
+template <size_t GROUPED_COLUMNS>
+std::vector<size_t>
+GroupBy::HashMapAggregationData<GROUPED_COLUMNS>::getHashEntries(
+    const T<std::span<const Id>> ids) {
+  AD_CONTRACT_CHECK(ids.size() > 0);
 
-  for (auto& val : ids) {
-    auto [iterator, wasAdded] = map_.try_emplace(val, getNumberOfGroups());
+  std::vector<size_t> hashEntries;
+  size_t numberOfEntries = ids.at(0).size();
+  hashEntries.reserve(numberOfEntries);
+
+  // TODO: We pass the `Id`s column-wise into this function, and then handle
+  //       them row-wise. Is there any advantage to this, or should we transform
+  //       the data into a row-wise format before passing it?
```

Comment:
The answer is:
measure it.
It might well be that it is very efficient for a small number of group by columns, because the compiler knows the size of the `std::array` and can then very efficiently unroll the loop. And you don't really have an alternative to changing to row-major ordering here for your hashmap.

The only real alternative would be to implement the hashmap not as
`map<Array, Index>`
But basically map<map<map<.....>>` for three columns.
There is a lot that could be tweaked here (maybe first do a check for only the first column, AND then have a fallback map etc.), 
But that is not required for now.
But what you should do, is benchmark this code vs. your old code, to see if there's any difference in the performance of the one-variable case.
And also maybe setup benchmarks for 2 variables and maybe 3 (you rarely have more of them). So that we at least know, where there's potential for improvement.

Let me know when you have those benchmarks.